### ISSUE TITLE
[Doc]: Document how correctness is verified

### ISSUE THREAD

**Issue by samos123** [vllm-project/tpu-inference#1510]
### 📚 The doc issue

we want to know how is numerical correctness being validated

(the process)

like benchmark, logit comparision

The current docs here don't shed much light on how correctness is validated for models: https://docs.vllm.ai/projects/tpu/en/latest/recommended_models_features/

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues and checked the [documentation page](https://github.com/vllm-project/tpu-inference/tree/main/docs).

**Comment by samos123** [comment]
tpu-inference runs an LM evals and ensures eval result is expected of the published eval numbers. This is done by running for example https://github.com/vllm-project/tpu-inference/blob/main/scripts/vllm/integration/test_accuracy.py

**Comment by vipannalla** [comment]
Verifying correctness for a new model  is a multi-step process. During bringup, the logits generated by the tpu implementation for some choosen prompts are compared with the golden logits for those input prompts generated from reference impl (eg. huggingface) on gpu/cpu.

Once model shows coherent outputs for few manual prompts, we run full set of eval benchmarks (MMLU, GSM8k etc..) and verify the scores against the ones published with the model release. For example, Qwen3-Coder-480B has MBPP, MBPP+, HumanEval, and HumanEval+ benchmarks run and the score are compared to be within an error margin -- https://docs.vllm.ai/projects/recipes/en/latest/Qwen/Qwen3-Coder-480B-A35B.html#evaluation.

We use lm-eval and Evalplus to run the correctness benchmarks. An example command to run HumanEval:

```
evalplus.evaluate \
--model "BCCard/Qwen3-Coder-480B-A35B-Instruct-FP8" \
--dataset humaneval \
--backend openai \
-base-url http://localhost:8000/v1 \
--greedy
```

Finally, we enable some of the eval benchmarks for each model in our CI pipeline and for nightly builds for regression detection.
